src/taipy/core/config/checkers/_migration_config_checker.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from taipy.config._config import _Config
+from taipy.config.checker._checkers._config_checker import _ConfigChecker
+from taipy.config.checker.issue_collector import IssueCollector
+
+from ..._version._version_manager import _VersionManager
+from ...exceptions import NonExistingVersion
+from ..migration_config import MigrationConfig
+
+
+class _MigrationConfigChecker(_ConfigChecker):
+    def __init__(self, config: _Config, collector: IssueCollector):
+        super().__init__(config, collector)
+
+    def _check(self) -> IssueCollector:
+        if migration_config := self._config._unique_sections.get(MigrationConfig.name):
+            self._check_if_entity_property_key_used_is_predefined(migration_config)
+            for source_version, migration_functions in migration_config.migration_fcts.items():
+                self._check_valid_version(source_version)
+                for config_id, migration_function in migration_functions.items():
+                    self._check_callable(source_version, config_id, migration_function)
+        return self._collector
+
+    def _check_callable(self, source_version, config_id, migration_function):
+        if not callable(migration_function):
+            self._error(
+                MigrationConfig._MIGRATION_FCTS_KEY,
+                migration_function,
+                f"The migration function of config `{config_id}` from version {source_version}"
+                f" must be populated with Callable value.",
+            )
+
+    def _check_valid_version(self, source_version):
+        try:
+            _VersionManager._replace_version_number(source_version)
+        except NonExistingVersion:
+            self._error(
+                MigrationConfig._MIGRATION_FCTS_KEY,
+                source_version,
+                "The source version for a migration function must be a valid version number.",
+            )
```

Comment:
In the newest commit, `replace_version_number(source)` is still used since the latest version maybe a production version.

This is subject to change, since if we want the user to provide the target or both source-target version, we will need this. If we only want the source, then we can remove this.

Reply:
In my mind, when adding a migration function, defining the version is mandatory. Whether it is the source_version or the target_version, or both.

So, in the checker, we want to ensure:
1. that the version is actually provided
2. that the version provided is a valid production version.

Are we aligned on this?

If yes, I still don't get the point. What is the reason for replacing the version provided by the user?
Am I missing something?

Reply:
We don't need to replace the version.